### DISCUSSION TITLE
Progress bar for upload

### DISCUSSION THREAD

**Discussion by Piste** [LeoKlaus/Paperparrot#97] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
When uploading a scan, it just shows some text and an animation of a few dots

**Describe the solution you'd like**
Show a circle that's getting filled or a progress bar to indicate how far along it's gone

**Comment by LeoKlaus** [comment]
Hi Piste,
first off, thanks for using Paperparrot and for your suggestions.

I've considered this but the current implementation doesn't really allow for a meaningful progress bar.

While it's possible to track the progress of the upload itself, the consumption progress is not available through the API.

I'm currently considering changing the upload flow to let users assign more attributes before uploading, this will need some more testing though.

**Comment by Piste** [comment]
I realize consumption is more of a wildcard. I'd be okay with the whole process stopping at upload done with the understanding that the Paperless server is now in charge of consumption and the app has done what it needed to do.
I'm definitely buying this app, now, knowing how responsive you are. this is so amazing!

**Comment by LeoKlaus** [comment]
> I'd be okay with the whole process stopping at upload done with the understanding that the Paperless server is now in charge of consumption and the app has done what it needed to do.

I would honestly much prefer not waiting for the consumption at all. The upload flow is something I haven't been happy with since the beginning but I haven't found time to completely redo it yet. 
I hope to find some time for this on the weekend, but for now, the 1.3 release is my main priority.

**Comment by LeoKlaus** [comment]
The upload process is much faster now, especially with large libraries.
I’ll mark this as closed for now, I’ll think about adding a progress bar in the future.
